src/configure/configure.ts

```diff
@@ -453,44 +455,46 @@ class Orchestrator {
         extensionVariables.templateServiceEnabled = false;
         var appropriatePipelines: PipelineTemplate[] = [];
 
-        if (!extensionVariables.templateServiceEnabled) {
-            var localPipelines = await vscode.window.withProgress(
-                { location: vscode.ProgressLocation.Notification, title: Messages.analyzingRepo },
-                () => templateHelper.analyzeRepoAndListAppropriatePipeline(
-                    this.inputs.sourceRepository.localPath,
-                    this.inputs.sourceRepository.repositoryProvider,
-                    repoAnalysisResult,
-                    this.inputs.pipelineConfiguration.params[constants.TargetResource])
-            );
-            appropriatePipelines = localPipelines;
-        } else {
-            var remotePipelines: PipelineTemplate[] = await vscode.window.withProgress(
-                { location: vscode.ProgressLocation.Notification, title: Messages.analyzingRepo },
-                () => templateHelper.analyzeRepoAndListAppropriatePipeline2(
-                    this.inputs.azureSession,
-                    this.inputs.sourceRepository.localPath,
-                    this.inputs.sourceRepository.repositoryProvider,
-                    repoAnalysisResult,
-                    this.inputs.pipelineConfiguration.params[constants.TargetResource])
-            );
-            appropriatePipelines = remotePipelines;
-        }
-        let pipelineMap = this.getMapOfUniqueLabels(appropriatePipelines);
-        let pipelineLabels = Array.from(pipelineMap.keys());
-
-        // TO:DO- Get applicable pipelines for the repo type and azure target type if target already selected
-        if (pipelineLabels.length > 1) {
-            var selectedOption = await this.controlProvider.showQuickPick(
-                constants.SelectPipelineTemplate,
-                pipelineLabels.map((pipeline) => { return { label: pipeline }; }),
-                { placeHolder: Messages.selectPipelineTemplate },
-                TelemetryKeys.PipelineTempateListCount);
-            //only label gets finalized, template isn't final yet
-            this.inputs.potentialTemplates = pipelineMap.get(selectedOption.label);
-        }
-        else {
-            this.inputs.potentialTemplates = pipelineMap.get(pipelineLabels[0]);
-        }
+        await telemetryHelper.executeFunctionWithTimeTelemetry(async () => {
```

Comment:
This is too big scope. We can put the timer only around making call to template service. Rest are local processing and will be very quick.

Reply:
updated

Reply:
okay, moving this to scope just the template service call